src/variorum/Nvidia/power_features.c

```diff
@@ -0,0 +1,241 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <power_features.h>
+#include <config_architecture.h>
+#include <variorum_error.h>
+#include <variorum_timers.h>
+
+void initNVML()
+{
+    unsigned int d;
+    /* Initialize GPU reading */
+    nvmlReturn_t result = nvmlInit();
+    nvmlDeviceGetCount(&m_total_unit_devices);
+    m_unit_devices_file_desc = (nvmlDevice_t*) malloc(sizeof(nvmlDevice_t) * m_total_unit_devices);
+
+    /* get handles to all devices */
+    for (d = 0; d < m_total_unit_devices; ++d)
+    {
+        int power;
+
+        result = nvmlDeviceGetHandleByIndex(d, &m_unit_devices_file_desc[d]);
+        if (result != NVML_SUCCESS)
+        {
+            printf("Error: Could not get GPU device handle.\n");
```

Comment:
There is a `variorum_error()` for common error reporting.

Reply:
Fixed.